api_tests/nodes/views/test_node_citations.py

```diff
@@ -1,80 +1,95 @@
-		assert_equal(res.status_code, 200)
-
-	def test_non_contributor_cannot_view_private_project_citations(self):
-		res = self.app.get(self.private_url, auth=self.non_contributor.auth, expect_errors=True)
-		assert_equal(res.status_code, 403)
-		assert_equal(res.json['errors'][0]['detail'], 'You do not have permission to perform this action.')
-
-	def test_unauthenticated_cannot_view_private_project_citations(self):
-		res = self.app.get(self.private_url, expect_errors=True)
-		assert_equal(res.status_code, 401)
-		assert_equal(res.json['errors'][0]['detail'], 'Authentication credentials were not provided.')
-
-	def test_unauthenticated_can_view_public_project_citations(self):
-		res = self.app.get(self.public_url)
-		assert_equal(res.status_code, 200)
-
-	def test_citations_are_read_only(self):
-		post_res = self.app.post_json_api(self.public_url, {}, auth=self.admin_contributor.auth, expect_errors=True)
-		assert_equal(post_res.status_code, 405)
-		put_res = self.app.put_json_api(self.public_url, {}, auth=self.admin_contributor.auth, expect_errors=True)
-		assert_equal(put_res.status_code, 405)
-		delete_res = self.app.delete_json_api(self.public_url, auth=self.admin_contributor.auth, expect_errors=True)
-		assert_equal(delete_res.status_code, 405)
-
-
-class TestNodeCitations(NodeCitationsMixin, ApiTestCase):
-
-	def setUp(self):
-		super(TestNodeCitations, self).setUp()
-		self.public_url = '/{}nodes/{}/citation/'.format(API_BASE, self.public_project._id)
-		self.private_url = '/{}nodes/{}/citation/'.format(API_BASE, self.private_project._id)
-
-
-class TestNodeCitationsStyle(NodeCitationsMixin, ApiTestCase):
-
-	def setUp(self):
-		super(TestNodeCitationsStyle, self).setUp()
-		self.public_url = '/{}nodes/{}/citation/apa/'.format(API_BASE, self.public_project._id)
-		self.private_url = '/{}nodes/{}/citation/apa/'.format(API_BASE, self.private_project._id)
+@pytest.fixture()
+def admin_contributor():
+    return AuthUserFactory()
+
+@pytest.fixture()
+def write_contrib():
+	return AuthUserFactory()
+
+@pytest.fixture()
+def read_contrib():
+	return AuthUserFactory()
+
+@pytest.fixture()
+def non_contrib():
+	return AuthUserFactory()
+
+@pytest.fixture()
+def public_project(admin_contributor):
+	return ProjectFactory(creator=admin_contributor, is_public=True)
+
+@pytest.fixture()
+def private_project(admin_contributor, write_contrib, read_contrib):
+	private_project = ProjectFactory(creator=admin_contributor)
+	private_project.add_contributor(write_contrib, permissions=['read','write'], auth=Auth(admin_contributor))
+	private_project.add_contributor(read_contrib, permissions=['read'], auth=Auth(admin_contributor))
+	private_project.save()
+	return private_project
+
+@pytest.mark.django_db
+class NodeCitationsMixin:
+
+    def test_node_citations(self, app, admin_contributor, write_contrib, read_contrib, non_contrib, private_url, public_url):
+
+	#   test_admin_can_view_private_project_citations
```

Comment:
This line is showing a flake error due to the indentation. Give these the correct indentation and spacing , e.g.

```python
def test_node_citations(self, app, admin_contributor, write_contrib, read_contrib, non_contrib, private_url, public_url):
    # test_blah_blah
    assert ....
```